### PR TITLE
Fixes homepage SVGs

### PR DESCRIPTION
**Pull request by naps62**
Why:

We were having weird problems with SVGs on Firefox. This could be anything
from a small typo in our code to a bug in Firefox itself. After hours of
debugging, we decided to move the SVGs to object elements, which seems to work
fine as well. There are extra requests being made now, but at least it's
working

This change addresses the need by:

* Moving all SVGs in the homepage to an `object` tag, instead of having them inline
* Refactor the JS code for the animations to interact with the contents of the object, instead of directly in the DOM

### CODE REVIEW

File: source/images/homepage/transparency.svg [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<svg class="LogoTransparency" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 302 302" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (azevedo-252) [inline source/images/homepage/transparency.svg]: I guess CSS doesn't reach this svg since it's inside an `<object>`

File: source/images/homepage/transparency.svg [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <path d="M320.44 147.12L38.658 316.15" stroke-linecap="square"/>
+            <path d="M320.44 113.12L38.658 282.15" stroke-linecap="square"/>
+            <path d="M-19.342 147.12L262.44 316.15" stroke-linecap="square"/>
+            <path d="M-19.342 113.12L262.44 282.15" stroke-linecap="square"/>
+            <path d="M150.06 309.144v-334.47"/>
+            <path d="M217.06 309.144v-334.47"/>
+            <path d="M83.06 309.144v-334.47"/>
+          </g>
+          <g stroke="url(#transparency-linearGradient-6)" stroke-width="2" mask="url(#transparency-mask-5)">
+            <path class="LogoTransparency-piece TransparencyIcon-pieceTop" d="M82 209.092l68.182 40.91 68.182-40.91v-36.364l-68.182 40.91L82 172.728v36.363z"/>
```
Comment (azevedo-252) [inline source/images/homepage/transparency.svg]: Is `.TransparencyIcon-pieceTop` being used for anything. I did a quick search and found no reference to it

File: source/javascripts/homepage/_lines.es6 [hunk 3]
```diff
@@ -108,4 +108,5 @@
 
   window.homepage = window.homepage || {};
   window.homepage.animateLines = animateLines;
-})(window);
+})(window)
+
```
Comment (azevedo-252) [inline source/javascripts/homepage/_lines.es6]: Not needed

File: source/javascripts/homepage/_logo_tools.es6 [hunk 4]
```diff
@@ -25,7 +26,8 @@ $(function() {
   }
 
   function animation(tool, options) {
-    var $tool = $(tool);
+    debugger
```
Comment (azevedo-252) [inline source/javascripts/homepage/_logo_tools.es6]: (╯°□°）╯︵ ┻━┻

File: source/javascripts/homepage/_logo_tools.es6 [hunk 5]
```diff
@@ -37,19 +39,19 @@ $(function() {
...
+    TweenLite.to($tool, 0.2, { rotation:  angle * direction, delay: delay });
+    TweenLite.to($tool, 0.4, { rotation: -angle * direction, delay: delay + 0.2 });
+    TweenLite.to($tool, 0.2, { rotation:      0,             delay: delay + 0.6, onComplete: callback });
   }
 
   function nameAnimation() {
-    TweenLite.set('.Logo-name', { fillOpacity: 0, x: 100 });
-    TweenLite.to('.Logo-name', 0.8, { x: 0, delay: 0.5 });
-    TweenLite.to('.Logo-name', 0.8, { fillOpacity: 1, delay: 0.5 });
+    TweenLite.set(svg.find('.Logo-name'), { fillOpacity: 0, x: 100 });
```
Comment (azevedo-252) [inline source/javascripts/homepage/_logo_tools.es6]: Maybe cache this? It's hitting the DOM for every search

File: source/stylesheets/all.scss [hunk 6]
```diff
@@ -4,3 +4,4 @@
 @import 'blue';
 @import 'utilities/base';
 @import 'components/base';
+
```
Comment (azevedo-252) [inline source/stylesheets/all.scss]: (╯°□°）╯︵ ┻━┻

File: source/stylesheets/components/_hover_animation.scss [hunk 7]
```diff
@@ -0,0 +1,4 @@
+.HoverAnimation {
+  pointer-events: all;
+}
+
```
Comment (azevedo-252) [inline source/stylesheets/components/_hover_animation.scss]: (╯°□°）╯︵ ┻━┻

File: source/stylesheets/components/_hover_animation.scss [hunk 8]
```diff
@@ -0,0 +1,4 @@
+.HoverAnimation {
```
Comment (azevedo-252) [inline source/stylesheets/components/_hover_animation.scss]: Can you provide documentation for this? Can you please state there why was this needed as well?

File: source/images/homepage/transparency.svg (hunk 1 above)
Comment (naps62) [inline source/images/homepage/transparency.svg]: CSS doesn't, but the JS animation code uses some of these classes to find elements

This particular one is not used, but it's children (`.LogoTransparency-piece`) still are. And I guess removing the parent while keeping the children wouldn't be right

File: source/images/homepage/transparency.svg (hunk 2 above)
Comment (naps62) [inline source/images/homepage/transparency.svg]: I guess you're right

File: source/javascripts/homepage/_lines.es6 (hunk 3 above)
Comment (naps62) [inline source/javascripts/homepage/_lines.es6]: fixed

File: source/javascripts/homepage/_logo_tools.es6 (hunk 4 above)
Comment (naps62) [inline source/javascripts/homepage/_logo_tools.es6]: ┬─┬ノ( º _ ºノ)

File: source/stylesheets/all.scss (hunk 6 above)
Comment (naps62) [inline source/stylesheets/all.scss]: ┬─┬ノ( º _ ºノ)

File: source/javascripts/homepage/_logo_tools.es6 (hunk 5 above)
Comment (naps62) [inline source/javascripts/homepage/_logo_tools.es6]: done

File: source/stylesheets/components/_hover_animation.scss (hunk 7 above)
Comment (naps62) [inline source/stylesheets/components/_hover_animation.scss]: ┬──┬﻿ ¯\_(ツ)

File: source/stylesheets/components/_hover_animation.scss (hunk 8 above)
Comment (naps62) [inline source/stylesheets/components/_hover_animation.scss]: done

Comment (azevedo-252) [inline source/stylesheets/components/_hover_animation.scss]: dont forget the styleguide section

File: source/stylesheets/components/_hover_animation.scss [hunk 9]
```diff
@@ -0,0 +1,13 @@
+// ContactsGrid
+//
+// Reacts to mouse events, allowing hover animations to execute
+// This was added due pointer-events being disabled by default for object elements.
```
Comment (azevedo-252) [inline source/stylesheets/components/_hover_animation.scss]: due **to**

Comment (naps62) [inline source/stylesheets/components/_hover_animation.scss]: fixed

File: source/stylesheets/components/_hover_animation.scss (hunk 8 above)
Comment (naps62) [inline source/stylesheets/components/_hover_animation.scss]: fixed

**Comment by azevedo-252** [comment]
:+1:
